axelor-sale/src/main/resources/domains/SaleOrderLineTree.xml

```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<domain-models xmlns="http://axelor.com/xml/ns/domain-models"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://axelor.com/xml/ns/domain-models http://axelor.com/xml/ns/domain-models/domain-models_6.1.xsd">
+
+  <module name="sale" package="com.axelor.apps.sale.db"/>
+
+  <entity name="SaleOrderLineTree">
+
+    <many-to-one name="saleOrderLine" title="Sale order line"
+      ref="com.axelor.apps.sale.db.SaleOrderLine" readonly="true"/>
+
+    <string name="code" title="Code" namecolumn="true" required="true"/>
+    <integer name="typeSelect" title="Type" selection="sale.sale.order.line.tree.select"/>
+    <many-to-one name="parent" title="Parent"
+      ref="com.axelor.apps.sale.db.SaleOrderLineTree" readonly="true"/>
+    <string name="title" title="Title" nullable="true"/>
+
+    <string name="description" title="Description" nullable="true"/>
+    <many-to-one name="product" title="Product" ref="com.axelor.apps.base.db.Product"/>
+    <decimal name="quantity" title="Quantity" default="1.00"/>
+    <many-to-one name="unit" title="Unit" ref="com.axelor.apps.base.db.Unit"/>
+    <decimal name="unitCost" title="Unit cost" nullable="true"/>
+    <decimal name="unitPrice" title="Unit price" nullable="true"/>
```

Comment:
I don't think it's possible to apply a dynamic scale on a tree view. So as we set the scale to 10 in the domain, it will look like this in the tree:
![image](https://github.com/axelor/axelor-open-suite/assets/31064803/9eb5932f-b667-4ad6-a699-7c90c0f7db63)